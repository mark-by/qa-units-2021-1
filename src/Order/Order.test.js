import React from 'react'
import {shallow, configure} from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import Order from "./Order";
import {fakeOrders} from "../data/fakeOrders";

jest.mock('../utils/getDate');
/* eslint-disable import/first */
import {getDate} from '../utils/getDate';

configure({adapter: new Adapter()});

describe('Order.js', () => {
    beforeEach(() => {
        getDate.mockReturnValue('11 марта, чт, 2021 год');
    })

    afterEach(() => {
        jest.resetAllMocks();
    })

    it('right render', () => {
        const wrapper = shallow(<Order order={fakeOrders[0]}/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('render without items', () => {
        const wrapper = shallow(<Order order={
            {
                id: 123,
                date: 1544356800000,
                shop: 'Alihandro Express',
            }
        }/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('render without shop', () => {
        const wrapper = shallow(<Order order={
            {
                id: 1,
                date: 1544356800000,
                items: [
                    'item1',
                    'item2'
                ]
            }
        }/>);
        expect(wrapper).toMatchSnapshot();
    });

    it('render without order', () => {
        const wrapper = shallow(<Order/>);
        expect(wrapper).toMatchSnapshot();
    });
});

