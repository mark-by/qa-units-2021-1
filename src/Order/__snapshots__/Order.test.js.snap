// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js render without items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      11 марта, чт, 2021 год
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order.js render without order 1`] = `""`;

exports[`Order.js render without shop 1`] = `""`;

exports[`Order.js right render 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      Alihandro Express
    </span>
    <span
      id="date"
    >
      11 марта, чт, 2021 год
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      Утиный пластмасса для показ новый год
    </div>
    <div
      className="Order-item"
      key="1"
    >
      Курица из нержавеющей стали, утка, гусь, голубь, питьевой фонтан
    </div>
    <div
      className="Order-item"
      key="2"
    >
      Новый стиль один розница яйцо для упаковки форма латекс
    </div>
  </div>
</div>
`;
